packages/health-and-info/src/index.ts

```diff
@@ -0,0 +1,195 @@
+import AgentOptions, { HttpOptions, HttpsAgent, HttpsOptions } from 'agentkeepalive'
+import superagent from 'superagent'
+import { Request, Response, NextFunction } from 'express'
+import {
+  ComponentCheckResult,
+  ExternalComponent,
+  CustomComponent,
+  ApplicationInfo,
+  HealthCheckOptions,
+  HealthCheckResult,
+} from './types/HealthCheckOptions'
+
+export default class HealthCheck {
+  private defaultOptions = {
+    components: {
+      retries: 2,
+      timeout: {
+        response: 1500,
+        deadline: 2000,
+      },
+      enabled: true,
+    },
+  }
+
+  constructor(
+    private readonly options: HealthCheckOptions,
+    private readonly logger: Console = console,
+  ) {}
+
+  private getDeploymentInfo = (appInfo: ApplicationInfo) => {
+    return {
+      build: {
+        buildNumber: appInfo.buildNumber,
+        gitRef: appInfo.gitRef,
+      },
+      version: appInfo.buildNumber,
+      uptime: Math.floor(process.uptime()),
+    }
+  }
+
+  private checkExternalComponents = (components: Partial<ExternalComponent | CustomComponent>[]) => {
+    return components
+      .filter(component => !('healthCheckFunction' in component))
+      .map(component => ({ ...this.defaultOptions.components, ...component }) as ExternalComponent)
+      .filter(component => component.enabled)
+      .map(this.checkExternalComponent)
+  }
+
+  private checkExternalComponent = async (component: ExternalComponent): Promise<ComponentCheckResult> => {
+    const { timeout, retries } = component
+    const agent = component.url.startsWith('https')
+      ? new HttpsAgent(component.agentConfig as HttpsOptions)
+      : new AgentOptions(component.agentConfig as HttpOptions)
+    let attemptsCount = 1
+
+    try {
+      const response = await superagent
+        .get(component.url)
+        .agent(agent)
+        .timeout(timeout)
+        .retry(retries, (err, res) => {
+          attemptsCount += 1
+          if (err) {
+            this.logger.info(
+              `Attempting to get health of external service ${component.name}, got: ${err.code} - ${err.message}`,
+            )
+          }
+        })
+
+      return {
+        name: component.name,
+        status: response.status === 200 ? 'UP' : 'DOWN',
+      }
+    } catch (error: unknown) {
+      this.logger.warn(
+        `Failed getting health of external service ${component.name}`,
+        (error as superagent.ResponseError).stack,
+      )
+
+      return {
+        name: component.name,
+        status: 'DOWN',
+        details: {
+          status: (error as superagent.ResponseError).status,
+          message: (error as superagent.ResponseError).message,
+          attempts: attemptsCount,
```

Comment:
**suggestion:** is this useful? if so is it worth adding it to the success response too?

Reply:
Attempts isn't the same as retries. Retries is just the amount of attempts done after the initial attempt failed, whereas attempts covers both the retries plus the initial attempt (so basically retries + 1). 

I think it could be useful adding it to the success response, considering not displaying it might be covering up the fact a service is only passing the health check after 5 attempts for example.